### PR TITLE
FAD-6025 IP Pool UI to support Enterprise tenants

### PR DESCRIPTION
**Pull request by rajumsys**
~- Add ip pools route flag under featureFlags~ This is handled by grants 
- Hide purchase cta for account with `ent1` plan
- Fixed current empty state bug

Notes:
- When purchase CTA is hidden, we should change this copy (on ip pools page)
> Purchase dedicated IPs to manage your IP Pools

Todo:

- [x] add tests for `currentPlanCodeSelector` as i exported it

### CODE REVIEW

File: src/config/default.js [hunk 1]
```diff
@@ -33,7 +33,8 @@ const config = {
   featureFlags: {
     allow_mailbox_verification: true,
     allow_anyone_at_verification: false,
-    has_signup: false
+    has_signup: false,
+    has_ip_pools: true
```
Comment (rajumsys) [inline src/config/default.js]: probably we won't need this at all as "grant" is already enough. i'm finding out about it.

File: src/pages/ipPools/ListPage.js [hunk 2]
```diff
@@ -67,25 +71,31 @@ export class IpPoolsList extends Component {
         primaryAction={createAction}
         title='IP Pools'
         empty={{
-          show: ipPools.length === 1 && ipPools[0].ips === 0,
+          show: ipPools.length === 1 && ipPools[0].ips.length === 0,
```
[the remark this answers is not in the document]
Comment (ewandennis) [reply src/pages/ipPools/ListPage.js]: Would you mind adding test coverage for the empty state so I can't ever re-introduce this bug ? 😄

File: src/pages/ipPools/ListPage.js [hunk 3]
```diff
@@ -67,25 +71,31 @@ export class IpPoolsList extends Component {
...
+          secondaryAction: showPurchaseCTA ? createAction : null,
+          primaryAction: showPurchaseCTA ? purchaseAction : createAction
         }}>
         { error ? this.renderError() : this.renderCollection() }
       </Page>
     );
   }
 }
 
+function shouldShowPurchaseCTA(state) {
```
[the remark this answers is not in the document]
Comment (rajumsys) [reply src/pages/ipPools/ListPage.js]: oh man! now i've to make `currentPlanCodeSelector` private and remove tests :P 

but yeah, not a bad idea

File: src/pages/ipPools/ListPage.js [hunk 4]
```diff
@@ -1,11 +1,15 @@
...
+import _ from 'lodash';
 import { connect } from 'react-redux';
 import { Link } from 'react-router-dom';
 import { listPools } from 'src/actions/ipPools';
 import { getOrderedIpPools } from 'src/selectors/ipPools';
+import { currentPlanCodeSelector } from 'src/selectors/accountBillingInfo';
 import { Loading, TableCollection, ApiErrorBanner } from 'src/components';
 import { Page } from '@sparkpost/matchbox';
 
+const plansWithDisallowedPurchaseCTA = ['ent1'];
```
[the remark this answers is not in the document]
Comment (rajumsys) [reply src/pages/ipPools/ListPage.js]: umm...i was not putting it config to avoid over-loading it unless we have a requirements to make it configurable. also didn't put in constants file as this is very specific to this page. i am fine with later (in constants file) though.

**Comment by rajumsys** [comment]
@ewandennis do you've any idea about the copy when purchase CTA is not showing?

File: src/pages/join/components/JoinForm.js [hunk 5]
```diff
@@ -72,6 +72,7 @@ export class JoinForm extends Component {
               name='first_name'
               component={TextFieldWrapper}
               label='First Name'
+              autoComplete='given-name'
```
Comment (rajumsys) [inline src/pages/join/components/JoinForm.js]: not related to this PR but thought it's tiny to solve here and make chrome happy

File: src/pages/ipPools/ListPage.js [hunk 6]
```diff
@@ -65,37 +61,33 @@ export class IpPoolsList extends Component {
 
     const createAction = { content: 'Create IP Pool', Component: Link, to: '/account/ip-pools/create' };
     const purchaseAction = { content: 'Purchase IPs', Component: Link, to: '/account/billing' };
+    const isEmptyState = ipPools.length === 1 && ipPools[0].ips.length === 0;
```
Comment (ewandennis) [inline src/pages/ipPools/ListPage.js]: Nice.

File: src/selectors/ipPools.js [hunk 7]
```diff
@@ -55,3 +58,11 @@ export const selectCurrentPoolInitialValues = createSelector(
     }, {})
   })
 );
+
+/**
+ * Returns whether Purchase IP cta should shown, based on account's plan code
+ * @return bool
+ */
+export const shouldShowIpPurchaseCTA = createSelector(
```
Comment (ewandennis) [inline src/selectors/ipPools.js]: A tiny bit of test coverage for this and we're done. 🎈
